### PR TITLE
Reduce timeout for waiting for resource version

### PR DESCRIPTION
**Pull request by wojtek-t**
Ref #37473

### CODE REVIEW

File: pkg/storage/watch_cache.go [hunk 1]
```diff
@@ -35,9 +35,10 @@ import (
 )
 
 const (
-	// MaximumListWait determines how long we're willing to wait for a
-	// list if a client specified a resource version in the future.
-	MaximumListWait = 60 * time.Second
+	// blockTimeout determines how long we're willing to block the request
+	// to wait for a given resource version to be propagated to cache,
+	// before terminating request and returning BadRequest error.
```
Comment (smarterclayton) [inline pkg/storage/watch_cache.go]: Is there a way to tie this to the users requested timeout?  It seems like if we can couple the timeout to users timeout we can reduce down to one knob to regulate

Comment (smarterclayton) [inline pkg/storage/watch_cache.go]: If we do a short timeout, we need to return 429 server timeout with a retry after.  That's the best way to put back pressure on clients that need this feature and want to be somewhat naive.

Comment (wojtek-t) [inline pkg/storage/watch_cache.go]: I thought about reusing request timeout, but in many cases it's not even set. And it's good to ensure that the request won't be pending forever.

I completely agree with the "retry-after" suggestion. Though I think we have a better solution than 429 - we have "504" (server timeout) that is also setting "retry after".

**Comment by wojtek-t** [comment]
@smarterclayton - PTAL

**Comment by wojtek-t** [comment]
@smarterclayton @liggitt - can you please take a look?

**Comment by smarterclayton** [comment]
Looks good, is there a test that covers this already (verifies the error type)?

**Comment by wojtek-t** [comment]
> Looks good, is there a test that covers this already (verifies the error type)?

No - will add one tomorrow.

**Comment by wojtek-t** [comment]
Added test. Applying label based on comment above.

**Pull request merged**